### ISSUE TITLE
Add ability to wait for specific message in the server streaming tests

### ISSUE THREAD

**Issue by alexromanov** [phiSgr/gatling-grpc#50]
**Is your feature request related to a problem? Please describe.**
I want to create a load test that will open a stream, get multiple messages until a message with a specific field will not be captured (within a defined timeout). (Without an explicit call of pause method and waiting for some time).

**Describe the solution you'd like**
Either add additional parameters for filtering messages and waiting for a timeout in the reconciliate method or add a specific message to the gRPC API of the plugin.

**Describe alternatives you've considered**
After an explicit call of pause() method, the reconciliation method is returning the last captured message. In our example case, it will return the last message - character 'E'. But the expected - is to filter message content and to stop when message content will be e.g. 'A'.

**Additional context**
Make sure, that when the reconciliation method with a filter will stop, Gatling will gracefully finish a stream and consider the request as a successful one. 

Sample [test](https://gist.github.com/alexromanov/90b96c5e31f33a28b677a47495585898)
Sample [server for test](https://github.com/alexromanov/server-grpc-sample)

**Comment by phiSgr** [comment]
Could you describe why exactly do you need this feature? Do you have a specific use case?

> will gracefully finish a stream
Do you want to cancel the server stream?

---

If I understand you correctly, what you want to do (other than the waiting) should be accomplished with:

```scala
.timestampExtractor { (_, message, streamStartTime) =>
  if (message.message == "E") streamStartTime else TimestampExtractor.IgnoreMessage
}
```

This will make Gatling write a successful request to the logs.

**Comment by phiSgr** [comment]
Other than not being able to wait for the stream to end in the main flow, does the following code mean what you have in mind?

```scala
.extract(_.state.some)(_ saveAs "state")
.timestampExtractor { (session, message, streamStartTime) =>
  if (session("state").asOption[State] == Some(State.DONE)) {
    throw new IllegalStateException("Message after DONE")
  }
  if (message.state == State.DONE) streamStartTime else TimestampExtractor.IgnoreMessage
}
```

**Comment by phiSgr** [comment]
> Single server stream request is resulting 4 stream responses.(Gatling is assuming each response as separate request)

> How can wait for all stream responses and only consider final response using TimestampExtractor.

As written in Scala doc:

```scala
 * @return The "start time" of this message,
 *         or [[TimestampExtractor.IgnoreMessage]] if this message should not be logged.
 */
def extractTimestamp(session: Session, message: Res, streamStartTime: Long): Long
```

**Comment by phiSgr** [comment]
This should be addressed in 0.12.0.
https://github.com/phiSgr/gatling-grpc/commit/94b4a3fbda7cb98874e95cbd9cbbd0b16a2d012d
